---
title: "C6103"
ms.custom: na
ms.date: "10/03/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: na
ms.suite: na
ms.technology: 
  - "vs-devops-test"
ms.tgt_pltfrm: na
ms.topic: "article"
f1_keywords: 
  - "C6103"
ms.assetid: 22d1ab35-31a3-4ba9-8ef4-7a64bce66621
caps.latest.revision: 4
ms.author: "douge"
manager: "douge"
translation.priority.ht: 
  - "cs-cz"
  - "de-de"
  - "es-es"
  - "fr-fr"
  - "it-it"
  - "ja-jp"
  - "ko-kr"
  - "pl-pl"
  - "pt-br"
  - "ru-ru"
  - "tr-tr"
  - "zh-cn"
  - "zh-tw"
---
# C6103
warning C6103: Returning \<variable> from failed function call at line \<location>.  
  
 A successful path through the function is returning a variable that was used as an `_Out_` parameter to an internal function call that failed.  
  
 The problem might be that the called function and the calling function are not completely annotated. The called function may require `_Always_`, `_Outptr_result_nullonfailure_` (`_COM_Outptr_` for COM code), or a related annotation, and the calling function may require a `_Success_` annotation. Another possible cause for this warning is that the `_Success_` annotation on the called function is incorrect.  
  
## See Also  
 [Using SAL Annotations to Reduce C/C++ Code Defects](../codequality/using-sal-annotations-to-reduce-c-c---code-defects.md)